src/Feed/SourceType/Brnd/ApiClient.php

```diff
@@ -0,0 +1,195 @@
+        } catch (\Throwable $throwable) {
+            throw new BrndException($throwable->getMessage(), (int) $throwable->getCode(), $throwable);
+        }
+    }
+
+    /**
+     * Get an authenticated scoped API client for the given FeedSource.
+     *
+     * @param FeedSource $feedSource
+     *
+     * @return HttpClientInterface
+     *
+     * @throws BrndException
+     */
+    private function getApiClient(FeedSource $feedSource): HttpClientInterface
+    {
+        $id = BrndFeedType::getIdKey($feedSource);
+
+        if (array_key_exists($id, $this->apiClients)) {
+            return $this->apiClients[$id];
+        }
+
+        $secrets = new SecretsDTO($feedSource);
+        $this->apiClients[$id] = HttpClient::createForBaseUri($secrets->apiBaseUri)->withOptions([
+            'headers' => [
+                'Content-Type' => 'application/json',
+                'Authorization' => 'Bearer '.$this->fetchToken($feedSource),
+                'Accept' => '*/*',
+            ],
+        ]);
+
+        return $this->apiClients[$id];
+    }
+
+    /**
+     * Get the auth token for the given FeedSource.
+     *
+     * @param FeedSource $feedSource
+     *
+     * @return string
+     *
+     * @throws BrndException
+     */
+    private function fetchToken(FeedSource $feedSource): string
+    {
+        $id = BrndFeedType::getIdKey($feedSource);
+
+        /** @var CacheItemInterface $cacheItem */
+        $cacheItem = $this->feedsCache->getItem('brnd_token_'.$id);
+
+        if ($cacheItem->isHit()) {
+            /** @var string $token */
+            $token = $cacheItem->get();
+        } else {
+            try {
+                $secrets = new SecretsDTO($feedSource);
+                $client = HttpClient::createForBaseUri($secrets->apiBaseUri);
+                $requestOptions = [
+                    'headers' => [
+                        'Content-Type' => 'application/json',
+                        'Accept' => '*/*',
+                    ],
+                    'json' => [
+                        'associationType' => self::AUTH_ASSOCIATION_TYPE,
+                        'apiAuthKey' => $secrets->apiAuthKey,
+                    ],
+                ];
+                $response = $client->request('POST', '/v1.0/generate-token', $requestOptions);
+
+                $content = $response->getContent(false); // Don't throw on non-2xx
+                $contentDecoded = json_decode($content, false, 512, JSON_THROW_ON_ERROR);
+                $token = $contentDecoded->data->access_token;
+
+                // Expire cache 5 min before token expire
+                $expireSeconds = intval(self::TOKEN_TTL - 300);
+                $cacheItem->set($token);
+                $cacheItem->expiresAfter($expireSeconds);
+                $this->feedsCache->save($cacheItem);
+            } catch (\Throwable $throwable) {
+                throw new BrndException($throwable->getMessage(), (int) $throwable->getCode(), $throwable);
```

Comment:
If you can throw some of the exceptions from src/Exceptions: https://github.com/os2display/display-api-service/tree/develop/src/Exceptions they will be mapped by Api Platform. See https://github.com/os2display/display-api-service/blob/develop/config/packages/api_platform.yaml#L55